File: easy/ruby/valid_palindrome.rb
# @param {String} s
# @return {Boolean}
def is_palindrome(s)
  s = s.downcase.gsub(/[^a-z0-9]/, '')

  left = 0
  right = s.length - 1

  while left < right
    return false if s[left] != s[right]

    left += 1
    right -= 1
  end

  true
end

File: easy/ts/valid_palindrome.ts
function isPalindrome(s: string): boolean {
  s = s.toLowerCase().replace(/[^0-9a-z]/g, '');
  let left = 0;
  let right = s.length - 1;
  while (left < right) {
    if (s[left] !== s[right]) {
      return false;
    } else {
      left++;
      right--;
    }
  }

  return true;
};